File: Genome Analyzer/API/GA/models/Staff.py
import sys
sys.path.append("..")
from GA import db
class Staff(db.Model):
    Name= db.Column(db.String(50),nullable=False)
    ID = db.Column(db.Integer, primary_key=True, nullable=True)
    Gender= db.Column(db.String(50),nullable=False)
    Phone= db.Column(db.Integer,nullable=True)
    Address= db.Column(db.String(50),nullable=True)
    Email= db.Column(db.String(50),nullable=True)
    Password= db.Column(db.String(50),nullable=True)
    Employee= db.Column(db.String(50),nullable=True)


    def __repr__(self):
        return f"Staff('{self.Name}', '{self.Gender}')"

File: Genome Analyzer/FrontEnd/src/modules/Staff.js
import axios from "axios";
export default {
  namespaced: true,
  state: {
    msg:""
  },
  mutations: {
    auth(state,res) {
      state.msg=res;
    },
  },
  actions: {
      addstaff({commit}, staff) {
        axios.defaults.headers.common["x-access-token"] = localStorage.getItem("access-token");
        console.log(localStorage.getItem("access-token"));
        console.log(staff);
        axios.post("https://genomicanalyzer.herokuapp.com/signup", {
          Name: staff.Name,
          Password: staff.Password,
          Gender: staff.Gender,
          Email: staff.Email,
          EmployeeType: staff.EmployeeType,
          Address: staff.Address,
          Phone: staff.Phone,
        })
        .then((response) => {
          console.log(response);
          commit("auth","success");
        })
        .catch(err=> {
          console.log(err);
          commit("auth","Some Thing wrong , Please try other user Name and all fied check is it not empty");
        });
    }
  }
  ,
  getters: {
    msg: (state) => state.msg,
  },
};
